src/components/Card.js

```diff
@@ -1,47 +1,96 @@
 export default class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
+  constructor(
+    { name, link, _id, isLiked },
+    cardSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleLike,
+    handleDislike
+  ) {
     this._name = name;
     this._link = link;
     this._cardSelector = cardSelector;
     this._cardElement = null;
+    this._id = _id;
+    this._isLiked = isLiked;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._handleLike = handleLike;
+    this._handleDislike = handleDislike;
+    this._handleLikeButtonClick = this._handleLikeButtonClick.bind(this);
+  }
+
+  _handleLikeButtonClick() {
+    if (this._isLiked) {
+      this._handleDislike(this);
+    } else {
+      this._handleLike(this);
+    }
   }
 
   _setEventListeners() {
     this._likeButton = this._cardElement.querySelector(".card__like-button");
     this._trashButton = this._cardElement.querySelector(".card__trash-button");
     this._cardImageEl = this._cardElement.querySelector(".card__image");
 
-    this._likeButton.addEventListener("click", this._handleLikeButtonClick);
-    this._trashButton.addEventListener("click", this._handleTrashButtonClick);
-    this._cardImageEl.addEventListener("click", () =>
-      this._handleImageClick({ name: this._name, link: this._link })
-    );
+    if (this._likeButton && this._trashButton && this._cardImageEl) {
+      this._likeButton.addEventListener("click", this._handleLikeButtonClick);
+      this._cardImageEl.addEventListener("click", () =>
+        this._handleImageClick({ name: this._name, link: this._link })
+      );
+      this._trashButton.addEventListener("click", () =>
+        this._handleDeleteClick(this)
+      );
+    } else {
+      console.error(
+        "Failed to set event listeners. One or more elements are undefined."
+      );
+    }
   }
-  _handleLikeButtonClick = () => {
-    this._likeButton.classList.toggle("card__like-button_active");
-  };
-
-  _handleTrashButtonClick = () => {
-    this._cardElement.remove();
-  };
-
-  getView() {
-    const cardTemplate = document
-      .querySelector(this._cardSelector)
-      .content.querySelector(".card");
 
-    this._cardElement = cardTemplate.cloneNode(true);
+  _updateLikeState() {
+    if (this._isLiked) {
+      this._likeButton.classList.add("card__like-button_active");
+    } else {
+      this._likeButton.classList.remove("card__like-button_active");
+    }
+  }
 
+  _updateView() {
     const cardTitleEl = this._cardElement.querySelector(".card__title");
     const cardImageEl = this._cardElement.querySelector(".card__image");
```

Comment:
You've found the image in line 34 and it's accessible here as `this._cardImageEl `

No need to search it again

Reply:
* You haven't corrected my remark https://github.com/Domvm-7/se_project_aroundtheus/pull/2#discussion_r1447495304